### ISSUE TITLE
Support Java 17

### ISSUE THREAD

**Issue by liff** [scalacenter/scalafix#1501]
Running the CLI on Java 17 results in:

```
java.lang.NoSuchFieldException: ucp
	at java.base/java.lang.Class.getDeclaredField(Class.java:2610)
	at scalafix.internal.reflect.ClasspathOps$.getURLs(ClasspathOps.scala:132)
	at scalafix.internal.reflect.ClasspathOps$.thisClassLoader(ClasspathOps.scala:38)
	at scalafix.internal.v1.Args$.$lessinit$greater$default$23(Args.scala:165)
	at scalafix.internal.v1.Args$.default(Args.scala:457)
	at scalafix.internal.v1.Args$.<clinit>(Args.scala:454)
	at scalafix.v1.Main$.run(Main.scala:34)
	at scalafix.v1.Main$.main(Main.scala:25)
	at scalafix.cli.Cli$.main(Cli.scala:9)
	at scalafix.cli.Cli.main(Cli.scala)
java.lang.NoSuchFieldException: ucp
	at java.base/java.lang.Class.getDeclaredField(Class.java:2610)
	at scalafix.internal.reflect.ClasspathOps$.getURLs(ClasspathOps.scala:132)
	at scalafix.internal.reflect.ClasspathOps$.thisClassLoader(ClasspathOps.scala:38)
	at scalafix.internal.v1.Args$.$lessinit$greater$default$23(Args.scala:165)
	at scalafix.internal.v1.Args$.default(Args.scala:457)
	at scalafix.v1.Main$.run(Main.scala:34)
	at scalafix.v1.Main$.main(Main.scala:25)
	at scalafix.cli.Cli$.main(Cli.scala:9)
	at scalafix.cli.Cli.main(Cli.scala)
java.lang.NoSuchFieldException: ucp
	at java.base/java.lang.Class.getDeclaredField(Class.java:2610)
	at scalafix.internal.reflect.ClasspathOps$.getURLs(ClasspathOps.scala:132)
	at scalafix.internal.reflect.ClasspathOps$.thisClassLoader(ClasspathOps.scala:38)
	at scalafix.v1.RuleDecoder$Settings$.apply(RuleDecoder.scala:220)
	at scalafix.internal.v1.Args.ruleDecoderSettings(Args.scala:247)
	at scalafix.internal.v1.Args.ruleDecoder(Args.scala:254)
	at scalafix.internal.v1.Args.configuredRules(Args.scala:288)
	at scalafix.internal.v1.Args.$anonfun$validate$1(Args.scala:420)
	at metaconfig.Configured.andThen(Configured.scala:45)
	at scalafix.internal.v1.Args.validate(Args.scala:416)
	at scalafix.internal.v1.MainOps$.run(MainOps.scala:72)
	at scalafix.v1.Main$.run(Main.scala:34)
	at scalafix.v1.Main$.main(Main.scala:25)
	at scalafix.cli.Cli$.main(Cli.scala:9)
	at scalafix.cli.Cli.main(Cli.scala)
```

It looks like the `ucp` field is no longer present in [`AppClassLoader` directly](https://github.com/openjdk/jdk17u/blob/2d69b2d5878ea0da361585e0253cb4ef13213290/src/java.base/share/classes/jdk/internal/loader/ClassLoaders.java#L169-L171) (vs. [Java 11](https://github.com/openjdk/jdk11u-dev/blob/c1411113b396f468963a1deacc3b57ed366e735a/src/java.base/share/classes/jdk/internal/loader/ClassLoaders.java#L157-L162)). It’s a field in [`BuiltinClassLoader`](https://github.com/openjdk/jdk17u/blob/2d69b2d5878ea0da361585e0253cb4ef13213290/src/java.base/share/classes/jdk/internal/loader/BuiltinClassLoader.java#L106-L107) though, so maybe it’d be possible to snatch the value from there somehow?

**Comment by tgodzik** [comment]
Thanks for reporting! I think we could use the same fix that in metals: https://github.com/scalameta/metals/blob/fe7e96fe67e765be279ad1ce61bc5eb4acac4cb8/mtags/src/main/scala/scala/meta/internal/mtags/ClasspathLoader.scala#L41

What do you think @mlachkar ?

**Comment by mlachkar** [comment]
Thanks! Yes we can try this fix!

**Comment by mnd999** [comment]
I'm also seeing this with Java 17:
```
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 61
	at scala.tools.asm.ClassReader.<init>(ClassReader.java:196)
	at scala.tools.asm.ClassReader.<init>(ClassReader.java:177)
	at scala.tools.asm.ClassReader.<init>(ClassReader.java:163)
	at scala.tools.asm.ClassReader.<init>(ClassReader.java:284)
	at scala.meta.internal.metacp.package$.scala$meta$internal$metacp$package$$readInputStreamToClassNode(package.scala:89)
	at scala.meta.internal.metacp.package$XtensionAsmClassfileOps.toClassNode(package.scala:75)
	at scala.meta.internal.symtab.GlobalSymbolTable.loadSymbol(GlobalSymbolTable.scala:37)
	at scala.meta.internal.symtab.GlobalSymbolTable.info(GlobalSymbolTable.scala:67)
	at scalafix.internal.v1.InternalSemanticDoc.info(InternalSemanticDoc.scala:75)
	at scalafix.internal.v0.LegacySemanticdbIndex.denotation(LegacySemanticdbIndex.scala:79)
	at scalafix.internal.v0.LegacySemanticdbIndex.$anonfun$denotation$2(LegacySemanticdbIndex.scala:84)
```

**Comment by mlachkar** [comment]
Thanks for reporting. I have just released the version 0.9.33 with java 17 support.

**Comment by mlachkar** [comment]
@liff can you please test that now it's fixed with 0.9.33 ? thanks

**Comment by liff** [comment]
0.9.33 on Java 17 is working fine for me, thanks.
